crates/sui-config/src/builder.rs

```diff
@@ -21,22 +21,42 @@ use sui_types::{
     },
 };
 
+pub enum CommitteeConfig {
+    Size(NonZeroUsize),
+    Validators(Vec<ValidatorGenesisInfo>),
+}
+
+enum ValidatorIpSelection {
+    Localhost,
+    Simulator,
+}
+
 pub struct ConfigBuilder<R = OsRng> {
-    rng: R,
+    rng: Option<R>,
     config_directory: PathBuf,
     randomize_ports: bool,
-    committee_size: NonZeroUsize,
+    committee: Option<CommitteeConfig>,
```

Comment:
I am tryin to understand why is this an option - seems like there is no code path in which it can be `None`?

Reply:
Actually, it was necessary - we need to be able to move out of `self.committee_config` in the `build()` method - the ValidatorGenesisInfo struct can't be made Clone() because it contains a keypair.